qiskit/primitives/backend_estimator_v2.py

```diff
@@ -162,7 +162,10 @@ def _run(self, pubs: list[EstimatorPub]) -> PrimitiveResult[PubResult]:
         pub_dict = defaultdict(list)
         # consolidate pubs with the same number of shots
         for i, pub in enumerate(pubs):
-            shots = int(math.ceil(1.0 / pub.precision**2))
+            data = self._preprocess_pub(pub)
+            bc_obs = data.observables
```

Comment:
Why not?
```suggestion
            bc_obs = pub.observables
```
It looks like you're doing a binding and then throwing it away. Why can't you just get the observables directly?